sdk/core/System.Net.ClientModel/tests/client/MapsClientTests.cs

```diff
@@ -26,4 +28,60 @@ public void TestClientSync()
         Assert.AreEqual("US", result.Value.CountryRegion.IsoCode);
         Assert.AreEqual(IPAddress.Parse("2001:4898:80e8:b::189"), result.Value.IpAddress);
     }
+
+    [Test]
+    public void TestCustomPipelineForMethodInvocation()
+    {
+        string key = Environment.GetEnvironmentVariable("MAPS_API_KEY");
+
+        KeyCredential credential = new KeyCredential(key);
+
+        // We have to pass the pipeline options to the client constructor
+        // so that the auth policy gets added, otherwise we won't have it later.
+        PipelineOptions pipelineOptions = new PipelineOptions();
+        pipelineOptions.IsLoggingEnabled = true;
+
+        MapsClient client = new MapsClient(new Uri("https://atlas.microsoft.com"), credential, pipelineOptions);
+
+        // Add a custom policy to the pipeline just for the one method
+        // This constructor is just a copy constructor;
+        // copies *settings* from pipeline options into request options
+
+        // Note: it does *not* copy the pipeline.
+        RequestOptions requestOptions = new RequestOptions(pipelineOptions);
```

Comment:
This method only copies pipelineOptions values into RequestOptions, it doesn't store the PipelineOptions

Reply:
Addressed